### ISSUE TITLE
rule: `@intlify/vue-i18n/prefer-sfc-locale-attr`

### ISSUE THREAD

**Issue by kazupon** [intlify/eslint-plugin-vue-i18n#241]
## Please describe what the rule should do:

enforce `locale` attr to be specified to custom block

## What category should the rule belong to?

- [x] Enforces code style (layout)
- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

## Provide 2-3 code examples that this rule should warn about

```vue
<i18n>
{
  "message": "hello!"
}
</i18n>
```

```vue
<i18n locale="en">
{
  "message": "hello!"
}
</i18n>

```

## Additional context

As a background, if we have built your own localization flow, and you use SFC i18n resources tool to collect it, if custom block content does not have a locale, it will not know which locale it is.

To avoid this issue, we can enforce the lint to specify the locale.

**Comment by ota-meshi** [comment]
Hmm. I think the rule also needs to check the JSON file, so I think there are a lot of false negatives when estimating the locale id to use from the placed file.
I want the user to set the locale ids to use if possible. I think it makes sense to allow common locale ids by default.
